flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -551,18 +579,47 @@ private boolean isSelfIgnore(final StateTransition<T> edge) {
 								nextVersion,
 								startTimestamp);
 					}
+
+					switch (skipStrategy.getStrategy()) {
+						case SKIP_PAST_LAST_ROW:
+							if (nextState.isFinal()) {
+								resultingComputationStates.add(createStartComputationState(computationState, outgoingEdges));
+							}
+							break;
+						case SKIP_TO_FIRST:
+							if (nextState.getName().equals(skipStrategy.getRpv()) &&
+								!nextState.getName().equals(currentState.getName())) {
+								ComputationState<T> startComputationState = createStartComputationState(computationState, outgoingEdges);
```

Comment:
Now I keep `startComputationState` instead of `startState` in NFA, so it can calculate the `outgoingEdges` from the start state when needed. Is this right?